detection_rules/devtools.py

```diff
@@ -391,25 +398,25 @@ def deprecate_rule(ctx: click.Context, rule_file: str):
 
     version_info = load_versions()
     rule_file = Path(rule_file)
-    contents = pytoml.loads(rule_file.read_text())
+    contents = TOMLRuleContents.from_dict(pytoml.loads(rule_file.read_text()))
     rule = TOMLRule(path=rule_file, contents=contents)
 
-    if rule.id not in version_info:
+    if rule.contents.id not in version_info:
         click.echo('Rule has not been version locked and so does not need to be deprecated. '
                    'Delete the file or update the maturity to `development` instead')
         ctx.exit()
 
     today = time.strftime('%Y/%m/%d')
 
-    new_meta = dataclasses.replace(rule.contents.metadata,
-                                   updated_date=today,
-                                   deprecation_date=today,
-                                   maturity='deprecated')
-    contents = dataclasses.replace(rule.contents, metadata=new_meta)
+    new_meta = {
```

Comment:
not a big fan of this change/function. it loses all type information, plus i'm not sure what the need/improvement was

Reply:
for clarification, are you referring to the `Rule.new` method? Can you describe what you want changed?

Reply:
basically instead of the `.new()` function, we should just explicitly call `replace`